src/transformers/tokenization_tapas.py

```diff
@@ -323,6 +326,823 @@ def save_vocabulary(self, save_directory: str, filename_prefix: Optional[str] =
+            max_length=max_length,
+            stride=stride,
+            pad_to_multiple_of=pad_to_multiple_of,
+            return_tensors=return_tensors,
+            prepend_batch_axis=True,
+            return_attention_mask=return_attention_mask,
+            return_token_type_ids=return_token_type_ids,
+            return_overflowing_tokens=return_overflowing_tokens,
+            return_special_tokens_mask=return_special_tokens_mask,
+            return_length=return_length,
+            verbose=verbose,
+        )
+
+        return BatchEncoding(batch_outputs)
+
+    def _batch_prepare_for_model(
+        self,
+        table_ids: List[int],
+        queries_ids: List[List[int]],
+        raw_table: pd.DataFrame,
+        raw_queries: Union[
+            List[TextInput],
+            List[PreTokenizedInput],
+            List[EncodedInput],
+        ],
+        answer_coordinates: Optional[List[Tuple]] = None,
+        answer_texts: Optional[List[TextInput]] = None,
+        add_special_tokens: bool = True,
+        padding: Union[bool, str, PaddingStrategy] = False,
+        truncation: Union[bool, str, TruncationStrategy] = False,
+        max_length: Optional[int] = None,
+        stride: int = 0,
+        is_split_into_words: bool = False,
+        pad_to_multiple_of: Optional[int] = None,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        return_token_type_ids: Optional[bool] = True,
+        return_attention_mask: Optional[bool] = True,
+        return_overflowing_tokens: bool = False,
+        return_special_tokens_mask: bool = False,
+        return_offsets_mapping: bool = False,
+        return_length: bool = False,
+        verbose: bool = True,
+        prepend_batch_axis: bool = False,
+        **kwargs
+    ) -> BatchEncoding:
+        """
+        Prepares a sequence of strings (queries) related to a table so that it can be used by the model. It creates
+        input ids, adds special tokens, truncates the table if overflowing (if the drop_rows_to_fit parameter is set to
+        True) while taking into account the special tokens and manages a moving window (with user defined stride) for
+        overflowing tokens
+
+        This function is based on prepare_for_model (but in Tapas, training examples depend on each other, so we
+        defined it at a batch level)
+
+        Args:
+            table: Pandas dataframe
+            queries: List of Strings, containing questions related to the table
+        """
+        batch_outputs = {}
+
+        if "table_data" in kwargs and "queries_tokens" in kwargs:
+            table_data = kwargs["table_data"]
+            queries_tokens = kwargs["queries_tokens"]
+        else:
+            table_data = None
+            queries_tokens = [None] * len(queries_ids)
+
+        for query_ids, raw_query, query_tokens, answer_coords, answer_text in zip(
+            queries_ids, raw_queries, queries_tokens, answer_coordinates, answer_texts
```

Comment:
If we change the answer_texts to answer_text, than we have to update this code:

```suggestion
        for query_ids, raw_query, query_tokens, answer_coords, answer_txt in zip(
            queries_ids, raw_queries, queries_tokens, answer_coordinates, answer_text
```

Reply:
Done